plugin/storage/integration/badgerstore_test.go

```diff
@@ -33,6 +33,10 @@ type BadgerIntegrationStorage struct {
 
 func (s *BadgerIntegrationStorage) initialize(t *testing.T) {
 	s.factory = badger.NewFactory()
+	t.Cleanup(func() {
+		s.factory.Close()
+	})
```

Comment:
Are you sure this needs to be right after NewFactory()? Usually the constructors NewFactory don't do any processing, and the actual work happens in `factory.Initialize`, so if `factory.Initialize` returns no error only then do you need to schedule a call to Close.

Reply:
yes, you are right. moved it after `factory.Initialize`.